wiki/Rules/Macro_rules/en.md

```diff
@@ -0,0 +1,75 @@
+# Macro rules
+
+Besides the rules outlaid here, as a general thumb rule it is generally not recommended to use any hardware or software adding unnecessary complexity or logic to your inputs. If you use a keyboard in the normal, intended way there is nothing to be afraid of.
+
+If you do use special features or anything the-like that you think could possibly be disallowed, it is worth reading the rules below.
+
+## What is considered a macro?
+
+Macroing is the utilization of automated procedures or specialized behaviour that alters your input from that of a typical, unmodified keyboard.
+The rules outlayed below affect both hardware and software-related functionality that alter game inputs in any way.
+
+![Flowchart](flowchart.png)
+
+Please note that the rules are specifically tailored towards keyboards as input devices. If you use anything else, do so at your own risk.
+
+## Glossary
+
+**Key**: A "key" refers to a physical key on your keyboard that can be moved upwards or downwards.
+
+**Key state**: A key state is the current state of a key as the osu! client perceives it. A key can be either be considered "pressed" or "released", regardless of the physical situation of the key.
+
+**Movement**: A movement is the continuous physical movement of a key in **one direction** (upwards or downwards). A direction change ends the movement and starts the next one. Whenever a key is "moved" in the rules, it refers to this definition.
+
+**Action**: An action is the act of the state of a key changing, regardless of what causes it.
```

Comment:
means nothing to me.

make it "game key" or "game action" like the original text.

Reply:
I've changed this and got completely rid of the "key state" thing, and I think this significantly improved the overall simplicity already